api.go

```diff
@@ -229,24 +275,30 @@ type tokenResponse struct {
 	Token    string
 }
 
+// message gets the first message of the supplised tokenResponse if present
+// otherwise default of "unknown" if tokenResponse was unsuccesfull
 func (r *tokenResponse) message() string {
 	if len(r.Messages) > 0 {
 		return r.Messages[0]
 	}
-	return "unknown"
+	if r.Success {
```

Comment:
We're hoping to remove the "success" field sometime soon and just use the HTTP status code. Please update the client code accordingly.